Fix issues with Robot Framework 6.1

bhirsz opened the pull request:
Not all issues are fixed but the fatal exception is. Remaining issues will be covered by branch https://github.com/MarketSquare/robotframework-robocop/compare/rf61-support or in additional PRs.

mnojek commented on docs/releasenotes/3.2.0.rst:
```diff
@@ -110,6 +110,16 @@ Possibly overwritten variables are now detected in more places (#841, #842)
+Fix for empty-lines-between-sections failing to parse empty section
+-------------------------------------------------------------------
+
+``empty-lines-between-sections`` rule should now work with empty sections with Robot Framework 6.1.
```
Would be good to add the rule id here.

mnojek commented on docs/releasenotes/3.2.0.rst:
```diff
@@ -110,6 +110,16 @@ Possibly overwritten variables are now detected in more places (#841, #842)
+Fix for empty-metadata not reported in Robot Framework 6.1
+-----------------------------------------------------------
+
+``empty-metadata`` rule should now work with Robot Framework 6.1.
```
Would be good to add the rule id here.

bhirsz commented:
@mnojek I have rebased your branch on top of my changes. I have also fixed several other issues (the goal is to have green build in this PR). Currently only parsing-error tests are failing - I should fix them today.

bhirsz commented:
Everything is fixed

bhirsz commented on robocop/checkers/errors.py:
```diff
@@ -282,7 +282,13 @@ class ParsingErrorChecker(VisitorChecker):
+        "is allowed only once. Only the first value is used",
+        "Test name cannot be empty",  # handled by test-case-name-is-empty
+        "User keyword name cannot be empty",  # handled by keyword-name-is-empty
+        "END is not allowed in this contex",  # handled by statement-outside-loop
```
Ah, it does not - and it works since we are checking "in error" and not equal. I will fix it

mnojek commented:
Thanks @bhirsz for this work! Massive amount of fixes and improvements 🔥
